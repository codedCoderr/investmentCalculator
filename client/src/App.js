import React, { useState } from 'react';
import axios from 'axios';
import { isNumber } from 'util';

function App() {
  const [subscription, setSubscription] = useState('');
  const [tenor, setTenor] = useState('');
  const [interest, setInterest] = useState('');
  const [investment, setInvestment] = useState('');
  const [error, setError] = useState('');
  const onSubscriptionChange = e => {
    setSubscription(e.target.value);
  };

  const onTenorChange = e => {
    setTenor(e.target.value);
  };
  const onInterestChange = e => {
    setInterest(e.target.value);
  };
  const onSubmit = async e => {
    try {
      e.preventDefault();
      const config = {
        headers: {
          'Content-Type': 'application/json'
        }
      };
      if (
        subscription !== isNumber ||
        tenor !== isNumber ||
        interest !== isNumber
      ) {
        setError('The values have to be numbers');
      }

      const data = JSON.stringify({ subscription, tenor, interest });
      const response = await axios.post(
        'http://localhost:3500/calculate',
        data,
        config
      );
      if (response.data) {
        setError('');
        setInvestment(response.data);
      }
    } catch (error) {
      // console.log(error);
    }
  };
  return (
    <div style={{ marginTop: '150px' }} className='App'>
      <article className='br3 ba b--black-10 mv4 w-100 w-50-m w-25-l mw6 shadow-5 center bg-hot-pink'>
        <main className='pa4 black-80'>
          <form className='measure'>
            <fieldset className='ba b--transparent ph0 mh0'>
              <legend className='f4 fw6 ph0 mh0'>Investment Calculator</legend>
              <div className='mt3'>
                <label className='db fw6 lh-copy f6' htmlFor='subscription'>
                  Monthly Subscription
                </label>
                <input
                  onChange={onSubscriptionChange}
                  className='pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100'
                  type='text'
                  subscription='subscription'
                  id='subscription'
                  autoComplete='off'
                  value={subscription}
                />
              </div>
              <div className='mt3'>
                <label className='db fw6 lh-copy f6' htmlFor='tenor'>
                  Tenor(Months)
                </label>
                <input
                  onChange={onTenorChange}
                  className='pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100'
                  type='tenor'
                  name='tenor'
                  id='tenor'
                  autoComplete='off'
                  value={tenor}
                />
              </div>
              <div className='mv3'>
                <label className='db fw6 lh-copy f6' htmlFor='interest'>
                  Interest Rate(Per annum)
                </label>
                <input
                  onChange={onInterestChange}
                  className='b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100'
                  type='interest'
                  name='interest'
                  id='interest'
                  autoComplete='off'
                  value={interest}
                />
              </div>
            </fieldset>
            <div className=''>
              <input
                onClick={onSubmit}
                className='b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib'
                type='submit'
                value='Calculate'
              />
              <div className=''>
                <br />
                {error}
              </div>
            </div>
          </form>
          <br />
          <div className=''>You should invest: {investment}</div>
        </main>
      </article>
    </div>
  );
}

export default App;
