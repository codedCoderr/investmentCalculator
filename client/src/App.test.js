import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const wrap = shallow(<App />);
const wrapper = toJson(wrap);

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
