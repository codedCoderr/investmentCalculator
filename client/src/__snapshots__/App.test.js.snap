// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div
  className="App"
  style={
    Object {
      "marginTop": "150px",
    }
  }
>
  <article
    className="br3 ba b--black-10 mv4 w-100 w-50-m w-25-l mw6 shadow-5 center bg-hot-pink"
  >
    <main
      className="pa4 black-80"
    >
      <form
        className="measure"
      >
        <fieldset
          className="ba b--transparent ph0 mh0"
        >
          <legend
            className="f4 fw6 ph0 mh0"
          >
            Investment Calculator
          </legend>
          <div
            className="mt3"
          >
            <label
              className="db fw6 lh-copy f6"
              htmlFor="subscription"
            >
              Monthly Subscription
            </label>
            <input
              autoComplete="off"
              className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
              id="subscription"
              onChange={[Function]}
              subscription="subscription"
              type="text"
              value=""
            />
          </div>
          <div
            className="mt3"
          >
            <label
              className="db fw6 lh-copy f6"
              htmlFor="tenor"
            >
              Tenor(Months)
            </label>
            <input
              autoComplete="off"
              className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
              id="tenor"
              name="tenor"
              onChange={[Function]}
              type="tenor"
              value=""
            />
          </div>
          <div
            className="mv3"
          >
            <label
              className="db fw6 lh-copy f6"
              htmlFor="interest"
            >
              Interest Rate(Per annum)
            </label>
            <input
              autoComplete="off"
              className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
              id="interest"
              name="interest"
              onChange={[Function]}
              type="interest"
              value=""
            />
          </div>
        </fieldset>
        <div
          className=""
        >
          <input
            className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib"
            onClick={[Function]}
            type="submit"
            value="Calculate"
          />
          <div
            className=""
          >
            <br />
          </div>
        </div>
      </form>
      <br />
      <div
        className=""
      >
        You should invest: 
      </div>
    </main>
  </article>
</div>
`;
